compiler/src/dmd/dclass.d

```diff
@@ -222,150 +223,155 @@ extern (C++) class ClassDeclaration : AggregateDeclaration
                 Type.typeinfofunction = this;
             }
             if (id == Id.TypeInfo_Delegate)
             {
                 if (!inObject)
-                    error("%s", msg.ptr);
+                    classError("%s `%s` %s", msg.ptr);
                 Type.typeinfodelegate = this;
             }
             if (id == Id.TypeInfo_Tuple)
             {
                 if (!inObject)
-                    error("%s", msg.ptr);
+                    classError("%s `%s` %s", msg.ptr);
                 Type.typeinfotypelist = this;
             }
             if (id == Id.TypeInfo_Const)
             {
                 if (!inObject)
-                    error("%s", msg.ptr);
+                    classError("%s `%s` %s", msg.ptr);
                 Type.typeinfoconst = this;
             }
             if (id == Id.TypeInfo_Invariant)
             {
                 if (!inObject)
-                    error("%s", msg.ptr);
+                    classError("%s `%s` %s", msg.ptr);
                 Type.typeinfoinvariant = this;
             }
             if (id == Id.TypeInfo_Shared)
             {
                 if (!inObject)
-                    error("%s", msg.ptr);
+                    classError("%s `%s` %s", msg.ptr);
                 Type.typeinfoshared = this;
             }
             if (id == Id.TypeInfo_Wild)
             {
                 if (!inObject)
-                    error("%s", msg.ptr);
+                    classError("%s `%s` %s", msg.ptr);
                 Type.typeinfowild = this;
             }
             if (id == Id.TypeInfo_Vector)
             {
                 if (!inObject)
-                    error("%s", msg.ptr);
+                    classError("%s `%s` %s", msg.ptr);
                 Type.typeinfovector = this;
             }
         }
 
         if (id == Id.Object)
         {
             if (!inObject)
-                error("%s", msg.ptr);
+                classError("%s `%s` %s", msg.ptr);
             object = this;
         }
 
         if (id == Id.Throwable)
         {
             if (!inObject)
-                error("%s", msg.ptr);
+                classError("%s `%s` %s", msg.ptr);
             throwable = this;
         }
         if (id == Id.Exception)
         {
             if (!inObject)
-                error("%s", msg.ptr);
+                classError("%s `%s` %s", msg.ptr);
             exception = this;
         }
         if (id == Id.Error)
         {
             if (!inObject)
-                error("%s", msg.ptr);
+                classError("%s `%s` %s", msg.ptr);
             errorException = this;
         }
         if (id == Id.cpp_type_info_ptr)
         {
             if (!inObject)
-                error("%s", msg.ptr);
+                classError("%s `%s` %s", msg.ptr);
             cpp_type_info_ptr = this;
         }
 
         baseok = Baseok.none;
     }
 
+    final void classError(const(char)* fmt, const(char)* arg)
```

Comment:
This function has a really awkward interface, with a format string that needs two extra `%s` arguments which aren't seen at the call site. You can easily deduplicate all `if (!inObject)` errors with a nested function, and spell out `kind, toPrettyChars` for the remaining few usages of `classError`.

Reply:
I'm not too happy about it, either, but this is just a transition stage to using ErrorSink, so it's not worth putting much effort into it.

We need to get error() out of Dsymbol, because that impairs the "dmd as library" initiative.